starknet_py/net/models/transaction.py

```diff
@@ -150,18 +218,49 @@ def calculate_hash(self, chain_id: StarknetChainId) -> int:
 
 
 @dataclass(frozen=True)
-class DeployAccount(AccountTransaction):
+class DeployAccountV3(_AccountTransactionV3):
+    """
+    Represents a transaction in the Starknet network that is a version 3 deployment of a Starknet account
+    contract.
+    """
+
+    class_hash: int
+    contract_address_salt: int
+    constructor_calldata: List[int]
+    type: TransactionType = TransactionType.DEPLOY_ACCOUNT
+
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        contract_address = compute_address(
+            salt=self.contract_address_salt,
+            class_hash=self.class_hash,
+            constructor_calldata=self.constructor_calldata,
+            deployer_address=0,
+        )
+        return compute_deploy_account_v3_transaction_hash(
+            class_hash=self.class_hash,
+            constructor_calldata=self.constructor_calldata,
+            contract_address_salt=self.contract_address_salt,
+            common_fields=self.get_common_fields(
+                tx_prefix=TransactionHashPrefix.DEPLOY_ACCOUNT,
+                address=contract_address,
+                chain_id=chain_id,
+            ),
+        )
+
+
+@dataclass(frozen=True)
+class DeployAccountV1(AccountTransaction):
     """
     Represents a transaction in the Starknet network that is a deployment of a Starknet account
     contract.
     """
 
+    max_fee: int = field(metadata={"marshmallow_field": Felt()})
```

Comment:
Could we do the same thing as in other places, create intermediate `DeprecatedAccountTransaction` and move `max_fee` there?

Reply:
First I saw [this comment](https://github.com/software-mansion/starknet.py/pull/1243/files#r1446337115). It sounds good to me.